vcd/resource_vcd_vm_internal_disk_test.go

```diff
@@ -0,0 +1,468 @@
+		"Allocated":                 "1024",
+		"Reserved":                  "1024",
+		"Limit":                     "1024",
+		"ProviderVdcStorageProfile": testConfig.VCD.ProviderVdc.StorageProfile,
+		// because vDC ignores empty values and use default
+		"MemoryGuaranteed": "1",
+		"CpuGuaranteed":    "1",
+
+		"Catalog":      testSuiteCatalogName,
+		"CatalogItem":  testSuiteCatalogOVAItem,
+		"VappName":     vappName,
+		"VmName":       vmName,
+		"ComputerName": vmName + "Unique",
+
+		"InternalDiskSize": internalDiskSize,
+	}
+	params["FuncName"] = t.Name() + "-IdeCreate"
+	configTextIde := templateFill(sourceTestVmInternalDiskIde, params)
+	params["FuncName"] = t.Name() + "-CreateALl"
+	configText := templateFill(sourceTestVmInternalDisk, params)
+	params["FuncName"] = t.Name() + "-Update1"
+	configText_update1 := templateFill(sourceTestVmInternalDisk_Update1, params)
+	params["FuncName"] = t.Name() + "-Update2"
+	//configText_update2 := templateFill(sourceTestVmInternalDisk_Update2, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText+configText_update1)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config:      configTextIde,
+				ExpectError: regexp.MustCompile(`.*The attempted operation cannot be performed in the current state \(Powered on\).*`),
+				Check: resource.ComposeTestCheckFunc(resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "size_in_mb", diskSize),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "bus_type", "ide"),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "bus_number", "0"),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "unit_number", "0"),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "storage_profile", storageProfile),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "allow_vm_reboot", "false"),
+					testCheckInternalDiskNonStringOutputs(internalDiskSize),
+				),
+			},
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName, "size_in_mb", diskSize),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName, "bus_type", busType),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName, "bus_number", busNumber),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName, "unit_number", unitNumber),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName, "storage_profile", storageProfile),
+					//resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName, "thin_provisioned", strconv.FormatBool(*adminVdc.IsThinProvision)),
```

Comment:
Is this still needed?

Reply:
will remove when I sure we finished changing design